lib/stbc.gi

```diff
@@ -398,12 +398,20 @@ end);
 InstallGlobalFunction(StabChainBaseStrongGenerators,function(arg)
 local   base,sgs,one,S,  T,  pnt, genlabels;
 
+    if not Length(arg) in [2, 3]
+        or not ForAll([1, 2], i -> IsHomogeneousList(arg[i])) then
+      ErrorNoReturn("usage: ",
+                    "StabChainBaseStrongGenerators(<base>, <sgs>[, <one>])");
+    fi;
     base:=PlainListCopy(arg[1]);
     sgs:=PlainListCopy(arg[2]);
     if Length(arg)=3 then
       one:=arg[3];
+    elif not IsEmpty(sgs) then
+      one:= One(sgs[1]);
     else
-      one:= One(arg[2][1]);
+      ErrorNoReturn("the identity element must be given as the third argument ",
+                    "when the second argument <sgs> is empty");
```

Comment:
But... why? Why don't we simply always do `one:= ()`?

Reply:
Good question, I have no idea. As I said at the top of this thread:
> (Can anyone think when you might want the one to not be ()?)

I can't think of a use case. Maybe @ChrisJefferson has an idea?

And @ssiccha – this function is documented to only apply to permutation groups, so it can't be because of semigroups.

Reply:
Sorry, I missed that question at the top 😊

Reply:
The only package calling `StabChainBaseStrongGenerators` is `grape` and it also passes `()` as third argument.

But in `TryPcgsPermGroup`, though, we pass `U.identity`... That gave me a clue: we have
```
lib/memory.gi:108:        S.identity := S.identity!.el;
```
So perhaps this is meant to allow supporting "permutations with memory" as produced by `GeneratorsWithMemory`?

Looking at `git blame`, the line in `TryPcgsPermGroup` was added by @hulpke 21 years ago in this commit:
```
Author:     Alexander Hulpke <hulpke@math.colostate.edu>
AuthorDate: 2000-01-25 18:50:09 +0000
Commit:     Alexander Hulpke <hulpke@math.colostate.edu>
CommitDate: 2000-01-25 18:50:09 +0000

    StabChainBaseStrongGenerators gets the appropriate `One' (this permits other
    permutation representations). SM&AH.
```
Perhaps he recalls something?

Reply:
If we use other representations (e.g. elements with memory, or straight line program elements) the identity element will be given in this representation.

Reply:
No intention to change it -- but clearly the documentation for this function then is wrong / incomplete

Reply:
Thanks for the explanation @hulpke, and don't worry, my purpose of the PR is to make the documentation match the behaviour of the code by changing the documentation, rather than to change the behaviour of the code.

However, I now feel a bit out of my depth, since I don't yet know anything about other representations for permutations in GAP.

I am also now worried that #4331 broken `StabChainBaseStrongGenerators` for your private code @hulpke – could you please check? In particular, the assertion `Assert(2, S.labels[1] = ());` is perhaps invalid, if a `one` other than `()` is specified.

Reply:
That should (apart from anyhow being in an assertion) be harmless as objects with memory would have to compare as equal to (). 
The reason for the different `one` is if elements get created that will be used later on. By starting with the correct `one' allos to build a proper word expression on the way.